src/slice_container.rs

```diff
@@ -0,0 +1,113 @@
+use std::{mem, slice};
+
+use ndarray::{ArrayBase, Dimension, OwnedRepr};
+use pyo3::class::impl_::{PyClassImpl, ThreadCheckerStub};
+use pyo3::pyclass::PyClass;
+use pyo3::pyclass_slots::PyClassDummySlot;
+use pyo3::type_object::{LazyStaticType, PyTypeInfo};
+use pyo3::{ffi, types::PyAny, PyCell};
+
+use crate::dtype::Element;
+
+/// Utility type to safely store Box<[_]> or Vec<_> on the Python heap
+pub(crate) struct PySliceContainer {
+    ptr: *mut u8,
```

Comment:
BTW, numpy does not allow mutating the array that has a parent array, so this can be `*const` in theory. So, let's try it if you are also curious if it works 😄

Reply:
I think owning raw pointers are `*mut` by convention.